### PR TITLE
gui/pkg_install_dialog: Allow license files to be automatically selected if it is already exist

### PR DESCRIPTION
**Pull request by nishinji**
This is useful when installing updates.

### CODE REVIEW

**Comment by Zangetsu38** [comment]
Should be good i look more later thx dalring 🥰😮🫡

File: vita3k/gui/src/pkg_install_dialog.cpp [hunk 1]
```diff
@@ -49,9 +49,24 @@ void draw_pkg_install_dialog(GuiState &gui, EmuEnvState &emuenv) {
...
+            fs::ifstream infile(pkg_path.string(), std::ios::binary);
+            PkgHeader pkg_header{};
+            infile.read(reinterpret_cast<char *>(&pkg_header), sizeof(PkgHeader));
+            infile.seekg(sizeof(PkgHeader));
+            std::string title_id_str(pkg_header.content_id);
+            std::string title_id = title_id_str.substr(7, 9);
+            const auto work_path{ fs::path(emuenv.pref_path) / fmt::format("ux0/license/{}/{}.rif", title_id, pkg_header.content_id) };
+            if (fs::exists(work_path)) {
+                LOG_INFO("Found license file: {}", work_path.string());
+                fs::ifstream binfile(work_path.wstring(), std::ios::in | std::ios::binary | std::ios::ate);
```
[the remark this answers is not in the document]
Comment (nishinji) [reply vita3k/gui/src/pkg_install_dialog.cpp]: Thanks.
I had used a code that was too old 👴

File: vita3k/gui/src/pkg_install_dialog.cpp [hunk 2]
```diff
@@ -48,9 +48,24 @@ void draw_pkg_install_dialog(GuiState &gui, EmuEnvState &emuenv) {
     if (draw_file_dialog) {
         result = host::dialog::filesystem::open_file(pkg_path, { { "PlayStation Store Downloaded Package", { "pkg" } } });
         draw_file_dialog = false;
-        if (result == host::dialog::filesystem::Result::SUCCESS)
-            ImGui::OpenPopup("install");
-        else if (result == host::dialog::filesystem::Result::CANCEL) {
+        if (result == host::dialog::filesystem::Result::SUCCESS) {
+            fs::ifstream infile(pkg_path.string(), std::ios::binary);
```
Comment (bookmist) [inline vita3k/gui/src/pkg_install_dialog.cpp:52]:
```suggestion
            fs::ifstream infile(pkg_path, std::ios::binary);
```
Here too. I forgot.

Comment (nishinji) [reply vita3k/gui/src/pkg_install_dialog.cpp:52]: 🤯

Comment (bookmist) [reply vita3k/gui/src/pkg_install_dialog.cpp]: Sorry, I was wrong. For filesystem.path it's better to use fs::path(pkg_path.native())
```
fs::ifstream infile(fs::path(pkg_path.native()), std::ios::binary);
//or
fs::ifstream infile(pkg_path.native(), std::ios::binary);
```
converting path to string can cause crash on some old systems (early versions of win10 and before) for unicode paths

File: vita3k/gui/src/pkg_install_dialog.cpp [hunk 3]
```diff
@@ -48,9 +48,24 @@ void draw_pkg_install_dialog(GuiState &gui, EmuEnvState &emuenv) {
...
         result = host::dialog::filesystem::open_file(pkg_path, { { "PlayStation Store Downloaded Package", { "pkg" } } });
         draw_file_dialog = false;
-        if (result == host::dialog::filesystem::Result::SUCCESS)
-            ImGui::OpenPopup("install");
-        else if (result == host::dialog::filesystem::Result::CANCEL) {
+        if (result == host::dialog::filesystem::Result::SUCCESS) {
+            fs::ifstream infile(pkg_path.native(), std::ios::binary);
+            PkgHeader pkg_header{};
+            infile.read(reinterpret_cast<char *>(&pkg_header), sizeof(PkgHeader));
+            infile.seekg(sizeof(PkgHeader));
```
Comment (bookmist) [inline vita3k/gui/src/pkg_install_dialog.cpp:55]: I can't find further use of `infile` and I'm not sure if this line is needed

Comment (nishinji) [reply vita3k/gui/src/pkg_install_dialog.cpp]: Indeed.
Removed.

**Review by bookmist — APPROVED** [review]

**Pull request merged**